### PR TITLE
feat: generic modal overlay

### LINKED ISSUE #383: Create generic modal container
**User Story:**

_As a user I want to view some page elements in an overlay in order to aid my understanding of the page hierarchy(!)._

**Product requirements/Acceptance criteria:**

Functional:
- The entire content of the modal (including search box) should scroll underneath the fixed banner.
- The modal should open and close using a fade transition.

### PR DESCRIPTION
**Pull request by tamlyn**
#### Background

This is a generic modal overlay container which should probably be contributed upstream to PubSweet. 

It handles animating in and out and scrolling of itself while locking scrolling of the parent page.

It makes no assumptions about layout or behaviour.

#### Any relevant tickets

Fixes #383.

#### How has this been tested?

Manually in the styleguide.

### CODE REVIEW

**Comment by hdrury1** [comment]
Hi @tamlyn, see slightly amended prod. recs for #346. As we're looking to re-use the people picker element of this outside of a modal context, this PR may well cover everything for that particular issue.

File: package.json [hunk 1]
```diff
@@ -88,7 +88,9 @@
     "react-dropzone": "^4.2.11",
     "react-hot-loader": "^4.1.2",
     "react-router-dom": "^4.2.2",
+    "react-scrolllock": "^3.0.1",
```
[the remark this answers is not in the document]
Comment (tamlyn) [reply package.json]: This prevents the underlying page from scrolling when the modal is displayed. It's a bit confusing if the part of the page you were on has scrolled away when you close the modal.

**Comment by tamlyn** [comment]
@hdrury1 thanks. Just the add and cancel buttons missing then. It seems that 'Add' is too specific to be added to the generic modal overlay component so perhaps that should be added specifically when the people picker is used in a modal context. I'll split this into a separate ticket. Do we have any other examples of modals or button bars in the app?

**Comment by hdrury1** [comment]
@tamlyn Yes, we'll be re-using the button-bar pattern later in the project (with different buttons, obviously) so it may be worth splitting this out?

**Comment by hdrury1** [comment]
@tamlyn Please see screenshot below for an example of where we will use the bar pattern again- in the decision control.

I won't share anything else of this with you yet, but see bar at the bottom of the right-hand pane.

screen shot 2018-07-20 at 11 09 16

**Pull request merged**